agent/virus_total_agent.py

```diff
@@ -42,26 +44,79 @@ def process(self, message: msg.Message) -> None:
             VirusTotalApiError: In case the Virus Total api encountered problems.
             NameError: In case the scans were not defined.
         """
-        response = virustotal.scan_file_from_message(message, self.api_key)
+        if message.data.get("content") is not None:
+            response = virustotal.scan_file_from_message(message, self.api_key)
+            self._process_response(response, message.data.get("path"))
+        else:
+            targets = self._prepare_targets(message)
+            for target in targets:
+                response = virustotal.scan_url_from_message(target, self.api_key)
+                self._process_response(response, target)
 
+    def _process_response(self, response: dict[str, Any], target: str | None):
         try:
             scans = virustotal.get_scans(response)
         except virustotal.VirusTotalApiError:
             logger.error("Virus Total API encountered some problems. Please try again.")
             raise
 
         try:
-            technical_detail = process_scans.get_technical_details(scans)
-            risk_rating = process_scans.get_risk_rating(scans)
-            self.report_vulnerability(
-                entry=kb.KB.VIRUSTOTAL_SCAN,
-                technical_detail=technical_detail,
-                risk_rating=risk_rating,
-            )
+            if scans is not None:
+                technical_detail = process_scans.get_technical_details(scans, target)
+                risk_rating = process_scans.get_risk_rating(scans)
+                self.report_vulnerability(
+                    entry=kb.KB.VIRUSTOTAL_SCAN,
+                    technical_detail=technical_detail,
```

Comment:
Are we sure we want to report empty technical_detail sometime,

Reply:
When will the technical details be empty?